task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -377,6 +379,7 @@ def parse(what: StartupDetails) -> RuntimeTaskInstance:
     bundle_instance = DagBundlesManager().get_bundle(
         name=bundle_info.name,
         version=bundle_info.version,
+        mark_in_use=True,
```

Comment:
we probably don't need to worry about protection re cleanup in between this checkout and task start